pkg/service/providerindex/legacy.go

```diff
@@ -21,25 +33,40 @@ type LegacyClaimsFinder interface {
 type LegacyClaimsStore struct {
 	contentToClaims ContentToClaimsMapper
 	claimsStore     types.ContentClaimsStore
+	claimsAddr      ma.Multiaddr
 }
 
 // ContentToClaimsMapper maps content hashes to claim cids
 type ContentToClaimsMapper interface {
 	GetClaims(ctx context.Context, contentHash multihash.Multihash) (claimsCids []cid.Cid, err error)
 }
 
-func NewLegacyClaimsStore(contentToClaimsMapper ContentToClaimsMapper, claimStore types.ContentClaimsStore) LegacyClaimsStore {
+func NewLegacyClaimsStore(contentToClaimsMapper ContentToClaimsMapper, claimStore types.ContentClaimsStore, claimsUrl string) (LegacyClaimsStore, error) {
+	legacyClaimsUrl, err := url.Parse(claimsUrl)
+	if err != nil {
+		return LegacyClaimsStore{}, err
+	}
+	claimsAddr, err := maurl.FromURL(legacyClaimsUrl)
+	if err != nil {
+		return LegacyClaimsStore{}, err
+	}
+
 	return LegacyClaimsStore{
 		contentToClaims: contentToClaimsMapper,
 		claimsStore:     claimStore,
-	}
+		claimsAddr:      claimsAddr,
+	}, nil
 }
 
 // Find looks for the corresponding claims for a given content hash in the mapper and then fetches the claims from the
 // claims store
 func (ls LegacyClaimsStore) Find(ctx context.Context, contentHash multihash.Multihash) ([]model.ProviderResult, error) {
 	claimsCids, err := ls.contentToClaims.GetClaims(ctx, contentHash)
 	if err != nil {
+		if errors.Is(err, types.ErrKeyNotFound) {
+			return []model.ProviderResult{}, nil
```

Comment:
Is...this...equivalent?
```suggestion
			return nil, nil
```

Reply:
from the perspective of code correctness, it is, as length checks, range and append operations, etc. will all behave as expected. However, from a readability standpoint, I think returning an empty slice is a clearer indication that an empty slice is a valid output of the function and we indeed mean to return that in that case.